Tests/Fixtures/Generated/IB-iOS/segues-swift3-context-all-ignore-module-need-extra-definitions.swift

```diff
@@ -0,0 +1,44 @@
+// swiftlint:disable all
+// Generated using SwiftGen, by O.Halligon — https://github.com/SwiftGen/SwiftGen
+
+// swiftlint:disable sorted_imports
+import Foundation
+import AVKit
+import CustomSegue
+import GLKit
+import LocationPicker
+import UIKit
+
+// swiftlint:disable superfluous_disable_command
+// swiftlint:disable file_length
+
+// MARK: - Storyboard Segues
+
+// swiftlint:disable explicit_type_interface identifier_name line_length type_body_length type_name
+internal enum StoryboardSegue {
+  internal enum AdditionalImport: String, SegueType {
+    case `private`
+  }
+  internal enum Message: String, SegueType {
+    case customBack = "CustomBack"
+    case embed = "Embed"
+    case nonCustom = "NonCustom"
+    case showNavCtrl = "Show-NavCtrl"
+  }
+  internal enum Wizard: String, SegueType {
+    case showPassword = "ShowPassword"
+  }
+}
+// swiftlint:enable explicit_type_interface identifier_name line_length type_body_length type_name
+
+// MARK: - Implementation Details
+
+internal protocol SegueType: RawRepresentable { }
+
+internal extension UIViewController {
+  func perform<S: SegueType>(segue: S, sender: Any? = nil) where S.RawValue == String {
+    performSegue(withIdentifier: segue.rawValue, sender: sender)
+  }
```

Comment:
Side note for another PR: I just realized that with our new possibility of generating multiple outputs from one single input — actually even now with the ability to parse multiple inputs, the problem might arise too — there's a possibility that a user would prefer generating multiple outputs declaring the same `protocol SegueType` and `extension UIViewController` etc… leading to compilation errors for multiple declarations.

So maybe for the next release we should think about a mechanism in our templates to ensure there's no repetition of our "Implementation Details" of the same template used multiple times. Should probably not be automatic (because depending on how the user will use the multiple output files they generate, maybe on the same target maybe not, they might want to have that implementation repeated on each output or not), but would probably be either via a `param` like `omitImplementationDeclarations` or `omitBoilerplate` or something, or having the Implementation details be declared in a separate, dedicated template (might be cleaner at implementation site with clearly separating that from the reusable templates, but that would require a lot of configuration in the `swiftgen.yml` file from the end user, so not cool. Or maybe using a `{% include %}` directive we could keep the boilerplate in a separate template for maintenance but conditionally auto-include it in the generated code depending on a `param`? Anyway, food for thoughts.

Reply:
Good idea, too bad we can't use the old ObjC trick to check if a macro is defined 😢 

I particularly like the `omitImplementationDeclarations`, but it's probably something for 6.1.